frontend/src/app/modules/work-space/dashboard/dashboard.component.ts

```diff
@@ -1,15 +1,61 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ProjectInfo } from '../../../shared/models/project-info';
+import { ToastrNotificationsService } from 'src/app/services/toastr-notifications.service';
+import { ProjectService } from '../../../core/services/project.service';
+import { User } from '@shared/models/user';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
-  styleUrls: ['./dashboard.component.sass']
+  styleUrls: ['./dashboard.component.sass'],
 })
-export class DashboardComponent implements OnInit {
+export class DashboardComponent implements OnInit, OnDestroy {
+  userProjects: ProjectInfo[];
+  cachedUserProjects: ProjectInfo[];
+  currentUser: User;
+  loading = false;
+  loadingProjects = false;
 
-  constructor() { }
+  private unsubscribe$ = new Subject<void>();
+
+  constructor(
+    private projectService: ProjectService,
+    private toastrService: ToastrNotificationsService
+  ) {}
 
   ngOnInit(): void {
+    this.getUserProjects(this.currentUser.id);
+  }
+
+  public ngOnDestroy(): void {
```

Comment:
Could you create a base component and implement ngOnDestroy there in order don't do it for every component again?

Reply:
As I understood, I have to create BaseComponent that implements OnDestroy. Do I need to add field "unsubscribe$" into BaseComponent and call methods next() and complete() in the ngOnDestroy()?

Reply:
yes, exactly that